docs(upgrade): restructure Pyth Core Upgrade section with per-chain pages (#3742)
pyth-network/pyth-crosschain: apps/developer-hub/content/docs/price-feeds/core/upgrade/preparing/index.mdx

---
title: "Preparing for the Pyth Core upgrade"
description: "Everything you need to upgrade your Pyth Core integration before July 31, 2026."
full: true
---

import { Button } from "@pythnetwork/component-library/Button";
import { Accordion, Accordions } from "fumadocs-ui/components/accordion";
import { Callout } from "fumadocs-ui/components/callout";
import { Card, Cards } from "fumadocs-ui/components/card";
import { Step, Steps } from "fumadocs-ui/components/steps";
import { Tab, Tabs } from "fumadocs-ui/components/tabs";
import {
  ActiveStepHighlighter,
  BranchSection,
  BranchToggle,
} from "../../../../../../src/components/MigrationFlow";

<ActiveStepHighlighter />

## What's changing

Pyth Network is upgrading [Pyth Core](/price-feeds/core) on **July 31, 2026**.
This upgrade replaces Pyth Core's underlying data infrastructure with an improved version, enabling new features:

- Higher-frequency updates
- Additional price feeds
- Lower latency

> - Customizable channels

Reviewer: I don't think channels really matter for Core; the on-chain contracts don't "see" channels and will accept data from any channel, so I'm not sure about mentioning it here.

| ---------------- | ------------------------------------------ | -------------------------------------------------------------------- |
| Timing           | You choose                                 | July 31, 2026                                                        |
| Downtime         | Up to you                                       | Brief, during the switch                                             |
| Hermes endpoint  | Switch to the new Hermes endpoint now             | Start using `hermes.pyth.network` with an API key before the cutover               |
| Contract address | You swap to the upgraded Pyth Core Contract       | DAO upgrades the current Pyth Core Contract for you                             |

<BranchSection path="now">

## Upgrading Now

This is the recommended path, as it makes the **July 31 cutover a non-event for your integration**. Please note that if you swap the endpoint but not the contract address or vice versa you won't be able to verify price updates. 

<Steps>
<Step>

### Move your Hermes calls to the new Hermes endpoint

> Switch your Hermes base URL from [`hermes.pyth.network`](https://hermes.pyth.network/docs/) to [`pyth.dourolabs.app/hermes`](https://pyth.dourolabs.app/docs/?urls.primaryName=Hermes+API) and add your API key that you got in Step 1. The two URLs serve the same data today. The upgraded endpoint requires authentication.

Reviewer: > The two URLs serve the same data today.

This reads like `hermes.pyth.network` and `pyth.dourolabs.app/hermes` serve the same price data which is just not true.

<Tabs items={["curl", "TypeScript SDK"]}>
<Tab value="curl">

```bash copy
curl -H "Authorization: Bearer $PYTH_API_KEY" \
  "https://pyth.dourolabs.app/hermes/v2/updates/price/latest?ids[]=0xe62df6c8b4a85fe1a67db44dc12de5db330f7ac66b72dc658afedf0f4a415b43"
```

</Tab>
<Tab value="TypeScript SDK">

```ts
import { HermesClient } from "@pythnetwork/hermes-client";

const client = new HermesClient("https://pyth.dourolabs.app/hermes", {
  accessToken: process.env.PYTH_API_KEY,
});
```

</Tab>
</Tabs>

Routes and response shapes are unchanged.
The upgraded endpoint is a drop-in replacement. See the new [Hermes API reference](https://pyth.dourolabs.app/docs/?urls.primaryName=Hermes+API#/) for the full surface.

<Callout type="warn">

>   **Deploy Step 2 and Step 3 in the same release.**

Reviewer: ```suggestion
  **Deploy 1 and 2 in the same release.**
```

  The upgraded endpoint serves payloads only the upgraded Pyth Core Contract verifies.
  If you ship these separately, **price updates can fail until both changes are live**.

>   If you cannot deploy both together, use the forward-compatible strategy in the Automatic Upgrade section.

Reviewer: ```suggestion
  If you cannot deploy both together, use a forward-compatible strategy like the one in the Automatic Upgrade section.
```
be less prescriptive as the strategy we show is EVM specific?

</Callout>

</Step>
<Step>

### Swap your contract address

Swap your existing Pyth contract address for the [upgraded Pyth Core Contract](/price-feeds/core/upgrade/contracts) on your chain. The upgraded Pyth Core Contract preserves the Pyth Core interface. No other code changes are needed.

View all upgraded [Pyth Core Contract addresses](/price-feeds/core/upgrade/contracts).

After the swap, you're done before the July 31 cutover. 

</Step>
</Steps>

### Per-chain guides

<Cards>
  <Card title="EVM" description="EVM-specific notes including ethers and viem usage." href="/price-feeds/core/upgrade/preparing/evm" />
  <Card title="Sui" description="Sui Move.toml change and package compatibility notes." href="/price-feeds/core/upgrade/preparing/sui" />
  <Card title="Solana" description="Solana Cargo feature flag and TS SDK configuration." href="/price-feeds/core/upgrade/preparing/solana" />
</Cards>

</BranchSection>

<BranchSection path="wait">

## Waiting for the Automatic Upgrade

At the cutover on July 31, the Pyth DAO upgrades your on-chain contract a few moments *before* `hermes.pyth.network` starts serving the upgraded payload.

> During that brief gap, the price updates serverd by `hermes.pyth.network` cannot verify against your upgraded contract, while those served by `pyth.dourolabs.app/hermes` can.

Reviewer: ```suggestion
During that brief gap, the price updates served by `hermes.pyth.network` cannot verify against your upgraded contract, while those served by `pyth.dourolabs.app/hermes` can.
```

  </Accordion>
  <Accordion title="What if my chain isn't supported?">
    Contact the team. Several chains are in active discussion and may be supported by July 31. For others, custom arrangements are possible. See the [upgraded Pyth Core Contract addresses](/price-feeds/core/upgrade/contracts) for the full list.
  </Accordion>
  <Accordion title="Are all my feeds available after the upgrade?">
    Nearly all current Pyth Core feeds remain available after the upgrade, with new ones added. Look up your specific feeds on the [feed explorer](https://pythdata.app/explore). If a feed you depend on isn't listed, [contact the team](#get-help).
  </Accordion>
  <Accordion title="I read prices on-chain only (push feeds, no Hermes) — do I need an API key?">
    No. The API key is only required for Hermes (REST/SSE) requests. If your integration reads prices directly from the Pyth contract on-chain and never calls Hermes, you don't need to register. Your contract will be upgraded automatically on July 31, or you can swap it early (Step 3).
  </Accordion>

>   <Accordion title="My code uses the Hermes REST/SSE API directly — what changes for me?">

Reviewer: on reading this section again, it sounds like it's for people not using HermesClient.
In that case it's fine to mention the HTTP header: Authorization: Bearer $PYTH_API_KEY

Also I found it strange that the - Automatic Upgrade item disappeared here

Author: I removed. I felt AUtomatic Upgrade was not necessary.

    Routes and response shapes are unchanged. The upgraded endpoint is a drop-in replacement. The differences are *which URL* you call and *whether you send an API key*:

>     - **Upgrade early (recommended):** switch your base URL to `pyth.dourolabs.app/hermes` and start sending the API key to your requests. The data is identical; only auth changes. See the [upgrade guide](/price-feeds/core/upgrade/preparing) for the full path.

Reviewer: ```suggestion
    - **Upgrade early (recommended):** switch your base URL to `pyth.dourolabs.app/hermes` and start sending the API key with your requests. The data format is identical; only auth changes. See the [upgrade guide](/price-feeds/core/upgrade/preparing) for the full path.
```

  </Accordion>
  <Accordion title="I read prices on-chain only (push feeds, no Hermes) — do I need an API key?">
    No. The API key is only required for Hermes (REST/SSE) requests. If your integration reads prices directly from the Pyth contract on-chain and never calls Hermes, you don't need to register. Your contract will be upgraded automatically on July 31, or you can swap it early (Step 3).
  </Accordion>
  <Accordion title="My code uses the Hermes REST/SSE API directly — what changes for me?">
    Routes and response shapes are unchanged. The upgraded endpoint is a drop-in replacement. The differences are *which URL* you call and *whether you send the `Authorization: Bearer $PYTH_API_KEY` header*:

    - **Upgrade early (recommended):** switch your base URL to `pyth.dourolabs.app/hermes` and start sending the `Authorization: Bearer $PYTH_API_KEY` header today. The data format is identical; only auth changes. See the [upgrade guide](/price-feeds/core/upgrade/preparing) for the full path.

>     - **Automatic upgrade:** keep calling `hermes.pyth.network` without the header today. At the cutover on July 31, push a deploy that adds the `Authorization: Bearer $PYTH_API_KEY` header on the same URL. See [Waiting for the Automatic Upgrade](/price-feeds/core/upgrade/preparing#waiting-for-the-automatic-upgrade) for the gap-handling pattern.

Author: That's wrong. I will fix it.
